src/Go/effects/netscriptGoImplementation.ts

```diff
@@ -322,6 +332,49 @@ export function getStats() {
   return statDetails;
 }
 
+export const boardValidity = {
+  valid: "",
+  badShape: "Invalid boardState: Board must be a square",
+  badType: "Invalid boardState: Board must be an array of strings",
+  badSize: "Invalid boardState: Board must be 5, 7, 9, 13, or 19 in size",
+  badCharacters:
+    'Invalid board state: unknown characters found. "X" represents black pieces, "O" white, "." empty points, and "#" offline nodes.',
+  failedToCreateBoard: "Invalid board state: Failed to create board",
+};
+
+/** Validate the given boardState */
+export function validateBoardState(error: (s: string) => void, _boardState: unknown): string[] | undefined {
+  if (!_boardState) {
+    return undefined;
+  }
+  if (!Array.isArray(_boardState)) {
+    error(boardValidity.badType);
+  }
+  if ((_boardState as unknown[]).find((row) => typeof row !== "string")) {
+    error(boardValidity.badType);
+  }
+
+  const boardState = _boardState as string[];
+
+  if (boardState.find((row) => row.length !== boardState.length)) {
+    error(boardValidity.badShape);
+  }
+  if (![5, 7, 9, 13, 19].includes(boardState.length)) {
+    error(boardValidity.badSize);
+  }
+  if (boardState.find((row) => row.match(/[^XO#.]/))) {
+    error(boardValidity.badCharacters);
+  }
+  try {
+    const board = boardFromSimpleBoard(boardState);
```

Comment:
All the places where you use this end up wanting the full board. Why not return the full board, and take that as the optional argument (on the internal, not external side only)?